### PR TITLE
Release-please

### PR DESCRIPTION
**Pull request by oldgiova**
Ticket: MC-7477

## Workflow
1. start from version 1.0.0 already deployed
2. The develop implemented (already) new `feat:` commits
3. `release-please` creates a new CHANGELOG PR on master, bumping the release to `1.1.0` and adding the CHANGELOG.md
4. New optional `feat:` or `fix:` commits would stay in the release `1.1.0` incrementing the CHANGELOG pr, until we're ready to release
5. When ready to release, merge the CHANGELOG.md PR
6. `release-please` creates a Github Release and a Github tag: v`1.1.0` on `master` branch
7. The release pipeline creates a PR to the `mender-docs-changelog` repo, adding the new CHANGELOG.md to the docs.md
8. The release pipeline run the `mender-dist-packages` pipeline with the variable `MENDER_APP_UPDATE_MODULE_VERSION: 1.1.0`
9. If you want you can create the branch `1.1.x` now.
10. If you create new features (`feat:`) or new breaking changes (`feat!:`), release-please will create new version `1.2.0` or `2.0.0` from `master`. You always can backport `fix:` commit to the `1.0.x` and `1.1.x` branches by cherry-picking them: the release pipeline will deploy the new releases and tags.

## Important note
Current Mender workflow is pretty out of standard: 
* main --> never release from main - doesn't have a proper version
* release branches (1.0.x, 1.1.x, 2.0.x etc..) --> you could release from a release branch

Release-please (and actually other tools) considers a different standard:
* main --> the rolling-release branch
* maintenance branches (1.0.x, 1.1.x, 2.0.x etc..) --> you could release from a release branch when you backport a fix

This second way is more strict to the SemVer, and it's the only way I can get the tool working properly.

### CODE REVIEW

**Comment by mzedel** [comment]
While I like the release-please approach of visualizing a release through a PR a lot, I am hesitant to use github releases as artifacts/ release reference due to the implications for our GH plan + the bulk of (enterprise) release artifacts that we won't be able to publish there... (+ I would vote for a custom changelog tool to make it easier to align with the previous changelog setup 🙃)

**Comment by oldgiova** [comment]
> I am hesitant to use github releases as artifacts/ release reference due to the implications for our GH plan + the bulk of (enterprise) release artifacts that we won't be able to publish there... 

Me too: this proposal is just for quickly visualizing releases, will not upload artifacts into them: the next step of the release process of this specific project is to run the `mender-dist-packages`, [automated as well](https://github.com/mendersoftware/app-update-module/pull/59/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R156-R165), which artifacts are not included in the GH releases.

> (+ I would vote for a custom changelog tool to make it easier to align with the previous changelog setup 🙃)
+1 for me: will work on it hopefully on the next iteration; this project is more of a greenfield, so we could start from an empty changelog with no previous standard defined.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -69,3 +96,70 @@ test:acceptancetests:
...
+    - hetzner-amd-beefy
+  rules:
+    - if: $CI_COMMIT_BRANCH =~ "/^\d+\.\d+\.x$/"
+    - if: $CI_COMMIT_BRANCH == $DEFAULT_BRANCH
+  script:
+    - npm install -g release-please
+    - release-please release-pr
+        --token=${GITHUB_BOT_TOKEN_REPO_FULL}
+        --repo-url=${GITHUB_REPO_URL}
+        --target-branch=${CI_COMMIT_REF_NAME}
```
Comment (mzedel) [inline .gitlab-ci.yml:113]: as a more general remark: IMO we could add the changelog customization here:
- generate changelog
- get the remote branch
- replace the release please changelog with the generated one
- push again

**Review by mzedel — APPROVED** [review]
If we run into a breaking change here, the commit linter and the release please changelog generation will cause some trouble, but IMO we can worry about this if/ once we actually get there.

File: .gitlab-ci.yml (hunk 1 above)
Comment (oldgiova) [reply .gitlab-ci.yml:131]: Thanks for the hint; I think I implemented the suggestions, please review again.

**Comment by oldgiova** [comment]
This is the resulting CHANGELOG.md from this `cliff.toml` conf: 
```
---
## 1.1.0 - 2024-09-16


### Features


- Oci images deep delta support ([MEN-7033](https://northerntech.atlassian.net/browse/MEN-7033))
- Automatic image extraction from docker-compose file 




### Testing


- Acceptance tests dind version bind 




## 1.0.0 - 2024-01-03


### Bug Fixes


- Bash test framework: cleanup phase, and misc. ([MEN-6366](https://northerntech.atlassian.net/browse/MEN-6366))
  misc fixes:
  * local variable (bt framework)
  * license in config file
  * misspell (in tests)
- Create modules directory if does not exist during install ([MEN-6196](https://northerntech.atlassian.net/browse/MEN-6196))
- Pass mode properly to install 
- DESTDIR and prefix support in the Makefile 
- Missing default declarations of `k8s_ctr_address` and `k8s_namespace` 




### Documentation


- *(compose)* Add documentation to compose configuration
- Basic readme content 




### Features


- App update module generator generator. ([MEN-6080](https://northerntech.atlassian.net/browse/MEN-6080))
- App Update Module. ([MEN-6076](https://northerntech.atlassian.net/browse/MEN-6076))
- Deep delta support: binary delta between image layers. ([MEN-6081](https://northerntech.atlassian.net/browse/MEN-6081)[MEN-6195](https://northerntech.atlassian.net/browse/MEN-6195))
  We can significantly reduce the binary delta size by taking
  into account corresponding layers inside the image.
  
  * add support for the generation of delta between layer.tar files
  * modify LOAD operation in the docker-compose orchestrator sub-module
- Bash test framework ([MEN-6082](https://northerntech.atlassian.net/browse/MEN-6082))
- Detecting composition errors and handling ([MEN-6367](https://northerntech.atlassian.net/browse/MEN-6367))
- Configuration for orchestrator sub modules ([MEN-6366](https://northerntech.atlassian.net/browse/MEN-6366))
- Requirements check. ([MEN-6368](https://northerntech.atlassian.net/browse/MEN-6368))
- K8s support ([MEN-6194](https://northerntech.atlassian.net/browse/MEN-6194))
- Cleanup, save, rollback. ([MEN-6077](https://northerntech.atlassian.net/browse/MEN-6077))




### Testing


- App-gen smoke test ([MEN-6079](https://northerntech.atlassian.net/browse/MEN-6079))
- Update module ArtifactInstall test. ([MEN-6079](https://northerntech.atlassian.net/browse/MEN-6079))
- Fixing the expected shasums mechanism 
- Acceptance test. ([MEN-6082](https://northerntech.atlassian.net/browse/MEN-6082))
- Verify that no pulling appeared ([MEN-6556](https://northerntech.atlassian.net/browse/MEN-6556))




### Refac


- *(app)* Refactor assert_requirements to only check status code




---
```

**Comment by mzedel** [comment]
Thank you very much for keeping at it @oldgiova!!

The only thing I would still adjust is putting the git-cliff config in a shared location (e.g. mendertesting) to ensure we have a consistent Changelog format across projects, but we can still extract this later IMO 👌.

**Comment by oldgiova** [comment]
> Thank you very much for keeping at it @oldgiova!!
> 
> The only thing I would still adjust is putting the git-cliff config in a shared location (e.g. mendertesting) to ensure we have a consistent Changelog format across projects, but we can still extract this later IMO 👌.

Agreed, making a note, but I'd prefer to do it in a second iteration: this way I can release the app-update-module and really test this thing in production.
Thanks and take care!

**Pull request merged**